cypress/integration/video-player.test.js

```diff
@@ -0,0 +1,113 @@
+import { phonePost, phoneListen, getAndClearLastPhoneMessage } from "../support";
+
+const authoredStateSample = {
+  version: 1,
+  prompt: "Test prompt",
+  videoUrl: "https://models-resources.s3.amazonaws.com/geniblocks/resources/fablevision/video/charcoal.mp4",
+  captionUrl: "https://models-resources.s3.amazonaws.com/question-interactives/test-captions.vtt",
+  poster: "https://models-resources.s3.amazonaws.com/geniblocks/resources/fablevision/rooms/missioncontrol.jpg",
+  credit: "Concord.org",
+  creditLink: "https://geniventure.concord.org",
+  creditLinkDisplayText: "Geniventure",
+  required: true
+};
+
+
+context("Test Video Player interactive", () => {
+  beforeEach(() => {
+    cy.visit("/wrapper.html?iframe=/video-player");
+  });
+
+  context("Runtime view", () => {
+
+    it("renders prompt and handles pre-existing interactive state", () => {
+        phonePost("initInteractive", {
+          mode: "runtime",
+          authoredState: authoredStateSample,
+          interactiveState: {
+            percentageViewed: 0.2,
+            lastViewedTimestamp: 1.2
+          }
+        });
+
+
+        cy.getIframeBody().find("#app").should("include.text", authoredStateSample.prompt);
+        cy.getIframeBody().find("#app").should("include.text", authoredStateSample.credit);
+        cy.getIframeBody().find("#app").should("include.text", authoredStateSample.creditLinkDisplayText);
+        // cy.getIframeBody().find("#app").find(".vjs-poster").should("have.attr", "style", `background-image: url("${authoredStateSample.poster}");`);
+        cy.getIframeBody().find("#app").find(".vjs-text-track-display").should("include.text", "This is a drake");
+        cy.getIframeBody().find("video").should("have.attr", "src", authoredStateSample.videoUrl);
```

Comment:
Can we check here if percentageViewed is respected somehow (is there some slider/progress bar that can be inspected easily)?